macos/file_picker/Sources/file_picker/FilePickerPlugin.swift

```diff
@@ -45,28 +52,32 @@ public class FilePickerPlugin: NSObject, FlutterPlugin {
     let extensions = args["allowedExtensions"] as? [String] ?? []
     applyExtensions(dialog, extensions)
 
-    if dialog.runModal() == NSApplication.ModalResponse.OK {
-      if allowMultiple {
-        let pathResult = dialog.urls
-        if pathResult.isEmpty {
-          result(nil)
+    if let appWindow = getFlutterWindow() {
+      dialog.beginSheetModal(for:appWindow) { response in
+        if (response == .OK) {
```

Comment:
There's some oddities with the control flow for the picker windows, due to not using `return` along with a `result(nil)` for unreachable cases. Can you verify that each branch calls the `result` block, so that we do not end up in situations where the result is never called? (that would cause the method call to hang on the Dart side)

I think that refactoring the handlers to use early returns instead of if-else, along with using `result(nil)` as a fallback should fix this. This is mostly due to the old implementation being written like that, though.
The new window let statement just makes it a bit harder to review in the diff :P

Reply:
I get what you mean! Now every execution branch gets the `result()` call. Sadly, using `guard let` did not help the PR readability.

Reply:
See my new comment for the readability thing :)